src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -175,20 +177,61 @@ public override void VisitDeclarationExpression(DeclarationExpressionSyntax node
         {
             var context = node?.Parent  // ArgumentSyntax
                               ?.Parent  // ArgumentListSyntax
-                              ?.Parent; // invocation/constructor initializer
+                              ?.Parent; // invocation/constructor initializer/index access, etc.
             switch (context?.Kind())
             {
                 case SyntaxKind.InvocationExpression:
                 case SyntaxKind.ObjectCreationExpression:
+                case SyntaxKind.ElementAccessExpression:
+                    {
+                        var local = SourceLocalSymbol.MakeVariableDeclaredInExpression(
+                            _scopeBinder.ContainingMemberOrLambda, _scopeBinder, _enclosingBinderOpt, node.Type(),
+                            node.Identifier(), LocalDeclarationKind.RegularVariable, (ExpressionSyntax)context);
+                        _localsBuilder.Add(local);
+                        break;
+                    }
+
                 case SyntaxKind.ThisConstructorInitializer:
                 case SyntaxKind.BaseConstructorInitializer:
-                    var local = SourceLocalSymbol.MakeOutVariable(_scopeBinder.ContainingMemberOrLambda, _scopeBinder, _enclosingBinderOpt, node.Type(), node.Identifier(), context);
-                    _localsBuilder.Add(local);
-                    break;
+                    {
+                        Debug.Assert(_enclosingBinderOpt == null || _enclosingBinderOpt == _scopeBinder);
+                        var local = SourceLocalSymbol.MakeOutVariableInCtorInitializer(
+                            _scopeBinder.ContainingMemberOrLambda, _scopeBinder, node.Type(), node.Identifier(),
+                            (ConstructorInitializerSyntax)context);
+                        _localsBuilder.Add(local);
+                        break;
+                    }
+
+                case SyntaxKind.ElementBindingExpression:
+                    {
+                        context = context.Parent;
```

Comment:
Actually, I will revise the PR to address this.

---
In reply to: [75947096](https://github.com/dotnet/roslyn/pull/13274#discussion_r75947096) [](ancestors = 75947096,75912066)